File: NeetCode/Two_Pointers/Two_Sum/two__sum.py
from typing import Optional, List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1

        while left < right:
            total = numbers[left] + numbers[right]
            if total == target:
                return [numbers[left], numbers[right]]
            elif total < target:
                left += 1
            else:
                right -= 1

        return None
        pass


if __name__ == "__main__":
    sol = Solution()
    print("Running Solution...")

File: NeetCode/Two_Pointers/Two_Sum/Two_Sum.cpp
// Two_Sum.cpp
#include <algorithm>
#include <array>
#include <cassert>
#include <cctype>
#include <climits>
#include <cmath>
#include <deque>
#include <functional>
#include <iostream>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>

using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        int left = 0;
        int right = numbers.size() - 1;
        int currSum = numbers[left] + numbers[right];
        while (currSum != target)
        {
            if (currSum < target)
            {
                left++;
            }
            else
            {
                right--;
            }
            currSum = numbers[left] + numbers[right];
        }
        return {left + 1, right + 1};
    }
};
